server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -2492,6 +2487,24 @@ protected void assertSnapshotOrGenericThread() {
                 }
             }
 
+            public NodeConnectionsService createTestNodeConnectionsService(
+                Settings settings,
+                ThreadPool threadPool,
+                TransportService transportService
+            ) {
+                return new NodeConnectionsService(settings, threadPool, transportService) {
+                    @Override
+                    public void connectToNodes(DiscoveryNodes discoveryNodes, Runnable onCompletion) {
+                        // just update targetsByNode to ensure disconnect runs for these nodes
```

Comment:
how is this different from `super.connectToNodes` ?

Reply:
we are only updating `targetsByNode` in the override and skipping all the actual connection logic

Reply:
Given that it is an existing test, why is this change required to skip connection logic now ?